File: Aditya-Pratap/Two Pointer/11.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n=len(height)
        i,j=0,n-1
        maxarea=0
        while i<j:
            h=min(height[i],height[j])
            b=j-i
            maxarea=max(h*b,maxarea)
            if height[i]<=height[j]:
                i+=1
            else:
                j-=1
        
        return maxarea

File: [Manya]/Leetcode/Two-Pointers/11.java
class Solution {
    public int maxArea(int[] height) {
      int MaxArea = 0;
      int left = 0;
      int right = height.length - 1;

      while(left < right){
        int currHeight = Math.min(height[left], height[right]);
        int area = (right - left) * currHeight;
        MaxArea = Math.max(MaxArea, area);

        if(height[left] < height[right]){
            left++;
        }
        else{
            right--;
        }
      }
      return MaxArea;
    }
}